src/eventer/eventer_jobq.c

```diff
@@ -414,6 +405,44 @@ eventer_jobq_enqueue(eventer_jobq_t *jobq, eventer_job_t *job, eventer_job_t *pa
   sem_post(&jobq->semaphore);
 }
 
+mtev_boolean
+eventer_jobq_try_enqueue(eventer_jobq_t *jobq, eventer_job_t *job, eventer_job_t *parent) {
+  job->next = NULL;
+  /* Do not increase the concurrency from zero for a noop */
+  if(ck_pr_load_32(&jobq->concurrency) == 0 && job->fd_event == NULL) {
+    free(job);
+    return mtev_true;
+  }
+  if(job->fd_event) {
+   if(jobq->max_backlog) {
+     /* Enforce a quick failure */
+     uint32_t bl;
+     do {
+       bl = ck_pr_load_32(&jobq->backlog);
+       if(bl >= jobq->max_backlog) {
+         free(job);
+         return mtev_false;
+       }
+     } while(!ck_pr_cas_32(&jobq->backlog, bl, bl+1));
+   }
+   else ck_pr_inc_32(&jobq->backlog);
```

Comment:
I don't see where the backlog is decremented.

Reply:
When the job "goes live"

```
  if(job) {
    job->next = NULL; /* To reduce any confusion */
    if(job->fd_event) ck_pr_dec_32(&jobq->backlog);
    ck_pr_inc_32(&jobq->inflight);
  }
```

Reply:
(this isn't new behavior).

In the original enqueue we upped total_jobs and backlog.

Now we split the increment so in the `internal` version only total_jobs is incremented and in the calling wrapper we're more careful w.r.t. incrementing backlog.